src/FormBuilder/resources/js/form-builder/src/components/canvas/FormSettingsContainer/components/SettingsGroup.tsx

```diff
@@ -0,0 +1,57 @@
+import {Children, isValidElement} from 'react';
+import classnames from 'classnames';
+import {__} from '@wordpress/i18n';
+import {chevronLeft, chevronRight} from '@wordpress/icons';
+import {useFormSettingsContext} from '@givewp/form-builder/components/canvas/FormSettingsContainer';
+import {
+    navigateBackInMenu,
+    updateMenuState,
+} from '@givewp/form-builder/components/canvas/FormSettingsContainer/formSettingsReducer';
+import {Icon} from '@wordpress/components';
+
+export default function SettingsGroup({item, title, children}) {
+    const [state, dispatch] = useFormSettingsContext();
+    const {content, activeMenu} = state;
+
+    const isActive = activeMenu === item;
+    const hasNestedMenu = Children.toArray(children).some((child) => {
+        return isValidElement(child) && child.type === SettingsGroup;
+    });
+
+    const handleItemClick = () => {
+        dispatch(updateMenuState(hasNestedMenu, item, children));
+    };
+
+    const handleBackClick = () => {
+        dispatch(navigateBackInMenu());
+    };
+
+    if (isActive && (!content || content !== children)) {
+        handleItemClick();
+    }
```

Comment:
@JasonTheAdams I've implemented a custom router system that I find much more reliable than the previous approach I used. I chose this solution because it involved a simple change to the existing code, without the need to include the entire React Router library.

https://github.com/impress-org/givewp/pull/7098/commits/df832e9fb66e90dbb3896a79d58e9451e3f3dca1